api/src/State/DownloaderProvider.php

```diff
@@ -10,10 +10,10 @@
 use App\Service\Downloader\DownloaderInterface;
 use Error;
 
-readonly class DownloaderProvider implements ProviderInterface
+class DownloaderProvider implements ProviderInterface
 {
     public function __construct(
-        private(set) DownloaderFactory $downloaderFactory
```

Comment:
This was not part of my request, do not make such changes please, please revert this line change